pkg/prometheus/statefulset.go

```diff
@@ -807,6 +808,7 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *operator.Config, shard in
 	}
 	if disableCompaction {
 		promArgs = append(promArgs, "--storage.tsdb.max-block-duration=2h")
+		promArgs = append(promArgs, "--storage.tsdb.min-block-duration=2h")
```

Comment:
this looks correct to me but leaving @brancz and/or @simonpasquier to have another look.

Reply:
The default value for `--storage.tsdb.min-block-duration` is already [2h](https://github.com/prometheus/prometheus/blob/063154eab720d8c3d495bd78312c0df090d0bf23/cmd/prometheus/main.go#L197-L198) and it shouldn't change for 2.x versions but it doesn't hurt to set it explicitly.